tools/clang/lib/SPIRV/EmitSPIRVAction.cpp

```diff
@@ -2063,7 +2063,138 @@ class SPIRVEmitter : public ASTConsumer {
     return 0;
   }
 
+  /// Returns true if the given CXXOperatorCallExpr is indexing into a vector or
+  /// matrix using operator[].
+  /// On success, writes the base vector/matrix into *base, and the indices into
+  /// *index0 and *index1, if there are two levels of indexing. If there is only
+  /// one level of indexing, writes the index into *index0 and nullptr into
+  /// *index1.
+  /// Assumes base, index0, and index1 are not nullptr.
+  bool isVecMatIndexing(const CXXOperatorCallExpr *vecIndexExpr,
+                        const Expr **base, const Expr **index0,
+                        const Expr **index1) {
+    // matrix [] []         vector []
```

Comment:
The comment here is a bit unclear and can be improved. e.g.

Is it `matrix[index0][index1]`, or `matrix[index1][index0]`

IIUC:
`vector = matrix[index0]`
and
`scalar = vector[index1]`